### DISCUSSION TITLE
entity-broadcast-range-percentage

### DISCUSSION THREAD

**Discussion by ColorsASD** [YouHaveTrouble/minecraft-optimization#114] [Show and tell]
On my server, due to the multitude of unique items, users have consistently experienced low FPS levels. While elsewhere they might have around 100-120 FPS, in my lobby, for example, it's around 50-60. You could try adjusting the "entity-broadcast-range-percentage" setting similarly to the "entity-tracking-range" setting. However, this setting affects everything that could cause lag and may also benefit those with poor internet connections. Here are a few visual examples with a value of 20:

![image](https://github.com/YouHaveTrouble/minecraft-optimization/assets/12263156/e6933041-da81-4d17-ac90-08b71c5b9fe2)
![image](https://github.com/YouHaveTrouble/minecraft-optimization/assets/12263156/aebbb24e-30e2-49bc-8785-6c669d6cce1b)

entity-broadcast-range-percentage=100 (63-65 FPS):
![image](https://github.com/YouHaveTrouble/minecraft-optimization/assets/12263156/78cc32b2-4c7d-4fbc-b275-545a642c0e23)

entity-broadcast-range-percentage=20: (74-78 FPS):
![image](https://github.com/YouHaveTrouble/minecraft-optimization/assets/12263156/b392ece5-3839-40ee-b93d-1d4ee402dc3b)

**Comment by ColorsASD** [comment]
My request/suggestion would be only to explore, with your settings, how and to what extent this setting could be connected to your preferences, so that they do not cause any conflicts.

**Comment by YouHaveTrouble** [comment]
As this is not really a server optimization matter, I'm gonna convert this to a discussion. This is still relevant for some, but generally if you have fps issues due to amount of entities, you probably have way too many entities.

**Comment by ColorsASD** [comment]
I made a mistake; this affects the "entity-tracking-range" setting in the spigot.yml file. Let's keep this value at 100%. We are talking about server-side performance, that's a fact, but we are doing it for the players, so that our server operates on a quality basis and there is no delay. This is equally true for FPS, especially when compared to other servers with 80% less delay.

**Reply by YouHaveTrouble** [reply]
> I made a mistake; this affects the "entity-tracking-range" setting in the spigot.yml file. Let's keep this value at 100%. We are talking about server-side performance, that's a fact

It's not, it's basically a list on the server side. If the entity is on the list, it's sent to the player, if not, it's not. It can be an issue on the client in terms of fps to track too many entities. It can also be networking problem if you don't have enough threads dedicated to netty relative to the player count or if you're tracking WAY too many entities (we're talking thousands within player's view distance). It is NOT a server side performance issue.

**Reply by ColorsASD** [reply]
I'm sorry that you can't appreciate how the server could be better with this setting, and that you don't understand what I'm writing. :/

**Comment by ColorsASD** [comment]
It only works bugged on the client side with armor stands and item frames; their disappearance doesn't occur when the Y coordinate moves. This is a bug on Mojang's part. However, it does react to changes in X and Z, which should be sufficient.
